### PR TITLE
Public count

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
```
Comment (dconeybe) [inline Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h:1]: Add `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END` to ALL of the new header files added in this directory. Then, remove all of the `_Nonnull` annotations. See other .h files in this directory for examples.

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * Executes the aggregate query and reads back the results as a `FIRAggregateQuerySnapshot`.
+ *
+ * @param source indicates where the results should be fetched from.
+ * @param completion a block to execute once the results have been successfully read.
+ *     snapshot will be `nil` only if error is `non-nil`.
+ */
+- (void)aggregationWithSource:(FIRAggregateSource)source
+                   completion:(void (^_Nonnull)(FIRAggregateQuerySnapshot *_Nullable snapshot,
+                                                NSError *_Nullable error))completion
+    NS_SWIFT_NAME(aggregation(source:completion:));
```
Comment (dconeybe) [inline Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h:42]: `aggregation` -> `getAggregation`

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+@interface FIRAggregateQuerySnapshot : NSObject
+
+/** The original `AggregateQuery` this snapshot is a result of. */
+@property(nonatomic, readonly) FIRAggregateQuery* _Nonnull query;
+
+/**
+ * The result of a document count aggregation. Null if no count aggregation is
+ *     available in the result.
+ */
+@property(nonatomic, readonly) NSNumber* _Nullable count;
```
Comment (dconeybe) [inline Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h:34]: Remove `_Nullable` from `count`.

**Review by dconeybe — CHANGES_REQUESTED** [review]

**Comment by dconeybe** [comment]
Add a change log entry to `Firestore/CHANGELOG.md`

e.g.
```
- [feature] Added `Query.count()`, which fetches the number of documents in the
  result set without actually downloading the documents (#10246).
```

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h (hunk 1 above)
Comment (wu-hui) [reply Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h:1]: Done.

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h (hunk 2 above)
Comment (wu-hui) [reply Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h]: Done.

**Comment by wu-hui** [comment]
> Add a change log entry to `Firestore/CHANGELOG.md`
> 
> e.g.
> 
> ```
> - [feature] Added `Query.count()`, which fetches the number of documents in the
>   result set without actually downloading the documents (#10246).
> ```

Done.

**Review by dconeybe — APPROVED** [review]

**Comment by dconeybe** [comment]
Note that this PR was reverted because it was accidentally merged prematurely: https://github.com/firebase/firebase-ios-sdk/pull/10252

**Pull request merged**